networkml/algorithms/host_footprint.py

```diff
@@ -110,6 +137,42 @@ def parse_args(raw_args=None):
         parsed_args = parser.parse_args(raw_args)
         return parsed_args
 
+    def _get_test_train_csv(self, path):
+        df = pd.read_csv(path)
+        df = self.reorder_drop_cols(df)
+        df = df.fillna(0)
+        # Split dataframe into X (the input features or predictors)
+        # and y (the target or outcome or dependent variable)
+        df['role'] = df.filename.str.split('-').str[0]
+        # Drop unknown roles.
+        df = df[df['role'] != 'Unknown']
```

Comment:
A question: Do we want the code to explicitly drop unknown? What if a user wanted to include a class named "unknown"? The code would drop this class and it would require some digging by the user for the user to understand that.

Reply:
OK I added a flag train_unknown (default False). It logs a message by default saying it's dropping role Unknown to let the user know.